doc/guidelines/e2e_frontend_coding_guidelines.md

```diff
@@ -0,0 +1,132 @@
+> In this case, you can locate the control by its associated label using [page.getByLabel()](https://playwright.dev/docs/api/class-page#page-get-by-label).
+
+Allows locating input elements by the text of the associated `<label>` or `aria-labelledby` element, or by the `aria-label` attribute.
+This is a reasonable alternative popular for input forms.
+
+### [getByText](https://playwright.dev/docs/locators#locate-by-text)
+
+> We recommend using text locators to find non interactive elements like div, span, p, etc. For interactive elements like button, a, input, etc. use [role locators](https://playwright.dev/docs/locators#locate-by-role).
+
+Allows locating elements that contain a given text.
+This alternative should be picked only if the other three are not possible.
+It is not very resilient and can frequently break the tests.
+
+### Others
+
+- getByPlaceholder() to locate an input by placeholder.
+- getByAltText() to locate an element, usually image, by its text alternative.
+- getByTitle() to locate an element by its title attribute.
+
+### [Playwright suggested locators](https://playwright.dev/docs/input)
+
+- text inputs: **label**
+- checkboxes: **label**
+- radio buttons: **label**
+- select options: **label**
+- mouse clicks: **role or text**
+- type characters: **locator**
+- keys and shortcuts: **text or role** **<- this is the only place where text is fine**
+- upload files and focus elements: **label**
+
+## 2. User facing or resilience?
+
+When chosing the right locator it is important to ask yourself one question.
+Do we want to test what the user perceives?
+If so, go with `roles`.
+If user-centricity is not so important on this element use `data-testid`.
+In case of doubt use `data-testid`.
+
+## 3. Naming Conventions
+
+#### Prefix
+
+We will add `test-` as a prefix to every `data-testid` element.
+
+#### Unique identifiers
+
+Well chosen names.
+
+```html
+<!-- Good -->
+<input type="text" data-testid="test-username-input" />
+
+<!-- Bad -->
+<input type="text" data-testid="test-input-1" />
+```
+
+#### Human readable
+
+No abbreviations.
+
+```html
+<!-- Good -->
+<button data-testid="test-login-button">Login</button>
+
+<!-- Bad -->
+<button data-testid="test-lgnbtn">Login</button>
+```
+
+### Resilient
+
+Chose a name that will remain relevant for longer time and avoid presentation details.
+
+```html
+<!-- Good -->
+<button data-testid="test-login-button">Login</button>
+
+<!-- Bad -->
+<button data-testid="test-tomato-login-button">Login</button>
+```
+
+### Be consistent
+
+Have a look at other id's around you.
+
+```html
+<!-- Good -->
+<input type="text" data-testid="test-username-input" />
+<input type="text" data-testid="test-password-input" />
+
+<!-- Bad -->
+<input type="text" data-testid="test-username-input" />
+<input type="text" data-testid="test-password" />
+```
+
+## 4. Incorporate Accessibility Best Practices
+
+Follow accessibility best practices while writing HTML code.
```

Comment:
This will be tricky or impossible for certain parts of the application, especially the editor.
Otherwise, I agree with this statement.

Reply:
The idea is that high accessible code is also easily testable. Of course it shouldn't be your main focus, but if you can achieve both in one go, that path should be taken I think.

Reply:
Yes. sounds like a plan, please also write something like that in the document.